### PR TITLE
[v2.9] Fix flaky nodescaling test cases + specify cluster types

### PR DESCRIPTION
**Pull request by markusewalker**
## Problem

During release testing, there were several issues seen with custom clusters and node driver for the `nodescaling` package. Additionally, there wasn't a way to see which cluster type is being ran for the `nodescaling` and `snapshot` package.

This makes it hard for Qase for reviewers to see exactly what cluster type was being ran against tests.
 
## Solution

- Addressed flakiness with nodescaling in counterpart shepherd PR with this
- Specified cluster types in test case names
 
### Automated Testing
- Ran against RKE1/RKE2/K3S for node scaling - custom cluster and node driver
- Ran snapshot test for each cluster type and made sure it is made clear in the test case header which cluster type is being used

### CODE REVIEW

File: tests/v2/validation/nodescaling/README.md
Comment (slickwarren) [inline tests/v2/validation/nodescaling/README.md:27]: since it depends if you're running rke2 or rke1, is there a way to specify that for the user? i.e. nodePools == rke1, machinePools == rke2/k3s?

File: tests/v2/validation/nodescaling/scale_replace_rke1_test.go
Comment (slickwarren) [inline tests/v2/validation/nodescaling/scale_replace_rke1_test.go:68]: supernit; the cluster type (rke1) is in the suite name, so we don't need it in the test name too.

File: tests/v2/validation/deleting/delete_cluster_rke1_test.go
Comment (slickwarren) [inline tests/v2/validation/deleting/delete_cluster_rke1_test.go:41]: another supernit here for having rke1 in the suite name and test name

File: tests/v2/validation/snapshot/snapshot_restore_k8s_upgrade_test.go
Comment (slickwarren) [inline tests/v2/validation/snapshot/snapshot_restore_k8s_upgrade_test.go:77]: just a note; funny we found different ways to get the k8s version lol. I used apiv1.clusterSpec -> cluster.Spec.

**Review by slickwarren — COMMENTED** [review]
just some small comments on the naming, everything else lgtm. waiting on jenkins jobs for official approval

File: tests/v2/validation/nodescaling/scale_replace_rke1_test.go
Comment (markusewalker) [reply tests/v2/validation/nodescaling/scale_replace_rke1_test.go:68]: True, I had it just there so it's the same across the board since it looked a bit cleaner to me. I can remove them though, not a big deal.

File: tests/v2/validation/nodescaling/README.md
Comment (markusewalker) [reply tests/v2/validation/nodescaling/README.md]: Resolved.

**Review by slickwarren — APPROVED** [review]

**Review by Josh-Diamond — APPROVED** [review]

**Review by Josh-Diamond — DISMISSED** [review]

File: tests/v2/validation/nodescaling/scale_replace_rke1_test.go
Comment (slickwarren) [inline tests/v2/validation/nodescaling/scale_replace_rke1_test.go:80]: nit; I don't like having the cluster type (rke1) in both the suite name and the test name(s).

File: tests/v2/validation/deleting/delete_cluster_test.go
Comment (slickwarren) [inline tests/v2/validation/deleting/delete_cluster_test.go:62]: couldn't you use similar logic for rke1 and remove the rke1 test file?

File: tests/v2/validation/nodescaling/scale_replace_test.go
Comment (slickwarren) [inline tests/v2/validation/nodescaling/scale_replace_test.go:93]: couldn't you use similar logic for rke1 and remove the rke1 test file?

File: tests/v2/validation/nodescaling/scaling_custom_cluster_rke1_test.go
Comment (slickwarren) [inline tests/v2/validation/nodescaling/scaling_custom_cluster_rke1_test.go:78]: supernit; might have asked this before but why have for 1 and for 2 and not just 2?

**Review by slickwarren — COMMENTED** [review]
some small comments to addres

File: tests/v2/validation/deleting/delete_cluster_test.go
Comment (markusewalker) [reply tests/v2/validation/deleting/delete_cluster_test.go:62]: That seems a bit excessive IMO, considering we already have a separate rke1 test file. I don't see why we wouldn't just keep it as it is.

File: tests/v2/validation/nodescaling/scale_replace_test.go
Comment (markusewalker) [reply tests/v2/validation/nodescaling/scale_replace_test.go:93]: That seems a bit excessive IMO, considering we already have a separate rke1 test file. I don't see why we wouldn't just keep it as it is.

File: tests/v2/validation/nodescaling/scaling_custom_cluster_rke1_test.go
Comment (markusewalker) [reply tests/v2/validation/nodescaling/scaling_custom_cluster_rke1_test.go:78]: Initially we agreed to have parity across node roles, but wanted to have at least one test case that added multiple nodes.

File: tests/v2/validation/nodescaling/scale_replace_test.go
Comment (markusewalker) [reply tests/v2/validation/nodescaling/scale_replace_test.go:93]: I see what you're saying. Considering shepherd PRs have already been merged, it's best practice to not make edits to the rancher/rancher PRs outside of just rebasing to update the shepherd version.

Let's look at that in a separate PR.

File: tests/v2/validation/nodescaling/scale_replace_rke1_test.go
Comment (markusewalker) [reply tests/v2/validation/nodescaling/scale_replace_rke1_test.go:80]: Let's look at that in a separate PR to consolidate the names altogether. The shepherd PRs have already been merged and it's best practice to not try make those kinds of edits now.

File: tests/v2/validation/nodescaling/scale_replace_test.go
Comment (slickwarren) [reply tests/v2/validation/nodescaling/scale_replace_test.go:93]: but this way, I have to run a completely different test if I want to run rke1. If its all in one file, its usually simpler for the tester. plus, less code/files to maintain

Comment (markusewalker) [reply tests/v2/validation/nodescaling/scale_replace_test.go:93]: ^ discussed further offline. Will do this in a separate PR shortly after this PR gets merged.

**Review by Josh-Diamond — APPROVED** [review]

**Review by slickwarren — APPROVED** [review]
these smaller concerns can be addressed in a separate PR since Shepherd has already merged.

**Pull request merged**
